test/parallel/test-stream-duplex-from.js

```diff
@@ -299,3 +299,20 @@ const { Blob } = require('buffer');
     assert.strictEqual(res, 'foobar');
   })).on('close', common.mustCall());
 }
+
+{
+  // Ensure piping to duplexified async iterator trigger the data call
+
+  const data = ['a', 'b', 'c', 'd'];
+
+  const cb = common.mustCall((chunk) => {
+    assert.strictEqual(chunk, data.shift());
+  }, data.length);
```

Comment:
Actually, this should be called only once, right?

Because the `highWaterMark` for the duplex stream that created from async generator function is 1:
https://github.com/nodejs/node/blob/61805fb9b427b332583a8061cfb755f9959cc17b/lib/internal/streams/from.js#L41

And according to the docs:
> The readable.pipe() method attaches a Writable stream to the readable, causing it to switch automatically into flowing mode and push all of its data to the attached Writable. **The flow of data will be automatically managed so that the destination Writable stream is not overwhelmed by a faster Readable stream**.
>
> [`readable.pipe(destination[, options])`](https://nodejs.org/api/stream.html#readablepipedestination-options)

and because no data callback exists here we reached the max buffer size.

WDYT @ronag?